### PR TITLE
Removed most Using Statements from CSharp

### PR DESCRIPTION
**Pull request by John-Hart**
Removed all Using Statements except for those with extension methods and Models

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@John-Hart__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by John-Hart** [comment]
This fixed #94 and #1109

**Comment by azurecla** [comment]
Hi __@John-Hart__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: src/generator/AutoRest.CSharp.Azure/TemplateModels/AzureMethodTemplateModel.cs [hunk 1]
```diff
@@ -278,14 +278,14 @@ public override string BuildUrl(string variableName)
 
         private void AddQueryParametersToUri(string variableName, IndentedStringBuilder builder)
         {
-            builder.AppendLine("List<string> _queryParameters = new List<string>();");
+            builder.AppendLine("System.Collections.Generic.List<string> _queryParameters = new System.Collections.Generic.List<string>();");
```
Comment (fearthecowboy) [inline src/generator/AutoRest.CSharp.Azure/TemplateModels/AzureMethodTemplateModel.cs]: If we use `var` for declarations wherever we can in places like this, 

``` csharp
   builder.AppendLine("var _queryParameters = new System.Collections.Generic.List<string>();");
```

we'll have less breakage later on if we change the type (and it cuts down on redundancy).

File: src/generator/AutoRest.CSharp/Templates/ExtensionsTemplate.cshtml [hunk 2]
```diff
@@ -5,14 +5,9 @@
 @EmptyLine
 namespace @Settings.Namespace
 {
-    using System;
-    using System.Collections;
-    using System.Collections.Generic;
-    using System.Threading;
     using System.Threading.Tasks;
```
Comment (fearthecowboy) [inline src/generator/AutoRest.CSharp/Templates/ExtensionsTemplate.cshtml]: What's preventing us from dropping `System.Threading.Tasks` ?

File: src/generator/AutoRest.CSharp.Azure/Templates/AzureMethodGroupTemplate.cshtml [hunk 3]
```diff
@@ -5,42 +5,31 @@
 @EmptyLine
 namespace @Settings.Namespace
 {
-    using System;
     using System.Linq;
```
Comment (fearthecowboy) [inline src/generator/AutoRest.CSharp.Azure/Templates/AzureMethodGroupTemplate.cshtml]: I notice that we've still got `System.Linq` still around... what's stopping us from dropping that?

File: src/generator/AutoRest.CSharp/Templates/MethodTemplate.cshtml [hunk 4]
```diff
@@ -93,18 +93,18 @@ public async Task<@(Model.OperationResponseReturnTypeString)> @(Model.Name)WithH
...
     // Tracing
-    bool _shouldTrace = ServiceClientTracing.IsEnabled;
+    bool _shouldTrace = Microsoft.Rest.ServiceClientTracing.IsEnabled;
     string _invocationId = null;
     if (_shouldTrace)
     {
-        _invocationId = ServiceClientTracing.NextInvocationId.ToString();
-        Dictionary<string, object> tracingParameters = new Dictionary<string, object>();
+        _invocationId = Microsoft.Rest.ServiceClientTracing.NextInvocationId.ToString();
+        System.Collections.Generic.Dictionary<string, object> tracingParameters = new System.Collections.Generic.Dictionary<string, object>();
```
Comment (fearthecowboy) [inline src/generator/AutoRest.CSharp/Templates/MethodTemplate.cshtml]: using `var` might be nice here too  :dancer:

File: src/generator/AutoRest.CSharp/Templates/MethodTemplate.cshtml [hunk 5]
```diff
@@ -212,12 +212,12 @@ else
             if (parameter.Type.IsPrimaryType(KnownPrimaryType.Stream))
             {
         <text>        
-        StreamContent _@parameter.Name = new StreamContent(@parameter.Name);
-        _@(parameter.Name).Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
+         System.Net.Http.StreamContent _@parameter.Name = new System.Net.Http.StreamContent(@parameter.Name);
```
Comment (fearthecowboy) [inline src/generator/AutoRest.CSharp/Templates/MethodTemplate.cshtml]: using `var` might be nice here too :boat:

File: src/generator/AutoRest.CSharp.Azure/Templates/AzureMethodGroupTemplate.cshtml (hunk 3 above)
Comment (John-Hart) [inline src/generator/AutoRest.CSharp.Azure/Templates/AzureMethodGroupTemplate.cshtml]: We still need the using any time there is extension methods

File: src/generator/AutoRest.CSharp/Templates/ExtensionsTemplate.cshtml (hunk 2 above)
Comment (John-Hart) [inline src/generator/AutoRest.CSharp/Templates/ExtensionsTemplate.cshtml]: Also has extension methods (i.e. UnWrap)

File: src/generator/AutoRest.CSharp.Azure/Templates/AzureMethodGroupTemplate.cshtml (hunk 3 above)
Comment (fearthecowboy) [inline src/generator/AutoRest.CSharp.Azure/Templates/AzureMethodGroupTemplate.cshtml]: Ah, yeah. 

The only option at that point would be to transform the call to a static method invocation. 

Which is kinda sucky, but not the end of the world.

File: src/generator/AutoRest.CSharp.Azure/AzureCSharpCodeGenerator.cs [hunk 6]
```diff
@@ -76,7 +76,7 @@ public override void NormalizeClientModel(ServiceClient serviceClient)
                     if (model.Extensions.ContainsKey(AzureExtensions.AzureResourceExtension) && 
                         (bool)model.Extensions[AzureExtensions.AzureResourceExtension])
                     {
-                        model.BaseModelType = new CompositeType { Name = "IResource", SerializedName = "IResource" };
+                        model.BaseModelType = new CompositeType { Name = "Microsoft.Rest.Azure.IResource", SerializedName = "Microsoft.Rest.Azure.IResource" };
```
Comment (fearthecowboy) [inline src/generator/AutoRest.CSharp.Azure/AzureCSharpCodeGenerator.cs]: (ignore this. testing a codeflow bug)

**Comment by fearthecowboy** [comment]
>                         model.BaseModelType = new CompositeType { Name = "IResource", SerializedName = "IResource" };

(ignore this too)

---
Refers to: src/generator/AutoRest.CSharp.Azure/AzureCSharpCodeGenerator.cs:79 in 5c413de. [](commit_id = 5c413de817b237c45ee0abc4d790605bb1fd37c0, deletion_comment = True)

**Comment by fearthecowboy** [comment]
LGTM :+1:

**Pull request merged**